### PR TITLE
Enable Traceability sidecars on testnet and mainnet

### LINKED ISSUE #4707: Enable Traceability sidecars, including Contract Action sidecar, in 0.35 on testnet and mainnet
Goal: Turn _on_ _all_ the Traceability sidecars in _all_ environments **except** leave contract actions **off* for testnet and mainnet.  Also, perform
the traceability _migration_ path in mainnet and _update_ the migration, already performed earlier, in testnet.

History:  A previous attempt to turn this on in `0.31.x` failed because it caused C2C spikes and thus increased avg C2C.  So it was reverted.  However, recent (late Dec 2022/early Jan 2023) performance testing on `0.34.0-alpha.0` and `.1` did _not_ show those spikes.  (Unknown yet whether this is degradation of avg C2C w/o those spikes.)  A lot has changed since `0.31.x` and we can't pin down why the spikes have gone.

So now, giving the positive evidence that there's no issue with the Action sidecar in 0.34.0-alpha we're going to turn on all sidecars in testnet and mainnet.  This will also enable the traceability migration in mainnet (that was already performed
in testnet - though it will be "updated" in testnet).

Desired _effective_ final state of properties in testnet and mainnet:

```
contracts.sidecars=CONTRACT_STATE_CHANGE,CONTRACT_BYTECODE
hedera.recordStream.enableTraceabilityMigration=true
```

### PR DESCRIPTION
**Pull request by david-bakin-sl**
Fixes #4707

Signed-off-by: David Bakin <117694041+david-bakin-sl@users.noreply.github.com>

**Description**:

Goal: Turn on Traceability sidecars for contract state and bytecode in _all_ environments, _and_ enable traceability migration in _all_ environments.  Also, turn on contract actions sidecar for all _except_ previewnet and mainnet.

**N.B.:** This is an updated newly-agreed plan: Turn on _only_ Contract Bytecode and Contract State in all environments; leave Contract Actions **off** in previewnet and mainnet.

History: A previous attempt to turn all sidecars - including contract actions - on in 0.31.x failed because it caused C2C spikes and thus increased avg C2C. So it was reverted. However, recent (late Dec 2022/early Jan 2023) performance testing on 0.34.0-alpha.0 and .1 did not show those spikes. (Unknown yet whether this is degradation of avg C2C w/o those spikes.) A lot has changed since 0.31.x and we can't pin down why the spikes have gone.

So now, giving the positive evidence that there's no issue with the Action sidecar in 0.34.0-alpha we're going to turn on all sidecars in testnet and mainnet.  This will also enable the traceability migration in mainnet (that was already performed
in testnet - though it will be "updated" in testnet).

**Effective** final state of properties in testnet and mainnet:

```
contracts.sidecars=CONTRACT_STATE_CHANGE,CONTRACT_BYTECODE
hedera.recordStream.enableTraceabilityMigration=true
```

The way `bootstrap.properties` works is that the file `hedera-node\hedera-mono-service/src/main/resources/bootstrap.properties` provides _all_ the values and then the `bootstrap.properties` in each "environment" directory under `hedera-node/configuration` has overrides.  

The result of this PR (as the following table shows) is to _set_ the proper values in the "default" file in `hedera-mono-service` and _remove_ these properties from all the individual environments thus giving _all_ environments the same
setup w.r.t. traceability sidecars.

||default|default|testnet|testnet|previewnet|previewnet|mainnet|mainnet|preprod|preprod|dev|dev|compose|compose|
|:--|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:| 
||before|after|before|after|before|after|before|after|before|after|before|after|before|after|
|`contracts.sidecars`|`SC,BC`|`SC,BC,CA`|-|`SC,BC`|-|-|`⎵`|`SC,BC`|-|-|-|-|-|
|`hedera.recordStream`<br>&nbsp;&nbsp;&nbsp;`.enableTraceabilityMigration`|`true`|`true`|-|-|-|-|`false`|-|`false`|-|-|-|-|-|

where: `-` means property entirely omitted from file, `⎵` means property present with _no_ value

N.B.: There are two more `bootstrap.properties` in the repo:

- `hedera-node/hedera-mono-service/src/test/resources/bootstrap.properties`
   - per Stoyan:
     > This property file reloads/reinitializes dynamic properties for the node when executing eet and integration tests with test clients. The node properties are updated several times during tests executions.
   - this file will be updated to _match_ the sidecar settings of the main default configuration
- `hedera-node/data/config/bootstrap.properties`
   - per Stoyan:
     > This file overrides the property values when running a local instance of hedera-node. IMO we don't need to override the defaults here.
    - this file already uses the defaults for sidecar configuration

**More changes not mentioned in the issue but in this PR:**

Property settings `hedera.recordStream.signatureFileVersion=6` and `hedera.recordStream.recordFileVersion=6` will be removed from all `bootstrap.properties` because they are set in the default configuration and those versions are used everywhere.

**Related issue(s)**:

Fixes #4707

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by david-bakin-sl** [comment]
||default|default|testnet|testnet|previewnet|previewnet|mainnet|mainnet|preprod|preprod|dev|dev|compose|compose|
|:--|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
||before|after|before|after|before|after|before|after|before|after|before|after|before|after|
|`contracts.sidecars`|`SC,BC,A`|`SC,BC,A`|`SC,BC,A`|`SC,BC,A`|
|`hedera.recordStream.enableTraceabilityMigration`|`true`|`true`|

**Review by stoqnkpL — APPROVED** [review]
LGTM!
I think it should be safe by now to also remove 
`hedera.recordStream.signatureFileVersion=6` & `hedera.recordStream.recordFileVersion=6`
 property overrides from all env folders as well. All environments are using version 6, right?

> What should the settings be for:
>hedera-node/hedera-mono-service/build/resources/{test,main}/bootstrap.properties

Gradle generates those and we don't need to update them.
>hedera-node/hedera-mono-service/src/test/resources/bootstrap.properties

This property file reloads/reinitializes dynamic properties for the node when executing eet and integration tests with test clients. The node properties are updated several times during tests executions. IMO it makes sense to have the default values here. This way, we can rely on CI to execute all tests with the values we'll be using for prod.
>hedera-node/data/config/bootstrap.properties

This file overrides the property values when running a local instance of hedera-node. IMO we don't need to override the defaults here.

File: hedera-node/configuration/previewnet/bootstrap.properties [hunk 1]
```diff
@@ -1,6 +1,3 @@
 ledger.id=0x02
 contracts.chainId=297
 bootstrap.genesisPublicKey=c249a323c878f5b5e2daccda6d731e6fdc32f870228d1cd4fae559d947dbc36c
-hedera.recordStream.recordFileVersion=6
-hedera.recordStream.signatureFileVersion=6
-contracts.sidecars=CONTRACT_STATE_CHANGE,CONTRACT_BYTECODE,CONTRACT_ACTION
```
Comment (Nana-EC) [inline hedera-node/configuration/previewnet/bootstrap.properties:6]: Please restore this.
We want to keep actions on in previewnet and local node for the mean time.

I think that affects the `preprod` and `dev/bootstrap.properties` also right?

**Review by Nana-EC — COMMENTED** [review]
We should maintain current preprod and previewnet settings that support actions right?

File: hedera-node/configuration/previewnet/bootstrap.properties (hunk 1 above)
Comment (david-bakin-sl) [reply hedera-node/configuration/previewnet/bootstrap.properties:6]: Per my table above (PR comment) preprod does **not** have actions turned on now.  Only previewnet and dev (whatever that is).  Also, AFAIK actions are also **not** on in local node which I believe has it's own (templating) mechanism for creating `bootstrap.properties` from a json file, and the key files over there, `configs/*.json` do not specify contract actions sidecar be enabled.

**Review by Nana-EC — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by stoqnkpL — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Comment by david-bakin-sl** [comment]
Turn on traceability sidecars and enable migration in all environments.

Mainnet and previewnet get only contract BYTECODE and STATE CHANGE; and do _not_ get the contract ACTION sidecar.  Other environments get all three.  All environments _now_ have traceability migration enabled (this affects only mainnet as the others have already had migration on.)

**Pull request merged**